clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1938,11 +1938,25 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
     if (Context.BuiltinInfo.isAuxBuiltinID(BuiltinID)) {
       assert(Context.getAuxTargetInfo() &&
              "Aux Target Builtin, but not an aux target?");
-
       if (CheckTSBuiltinFunctionCall(
               Context.getAuxTargetInfo()->getTriple().getArch(),
-              Context.BuiltinInfo.getAuxBuiltinID(BuiltinID), TheCall))
+              Context.BuiltinInfo.getAuxBuiltinID(BuiltinID), TheCall)) {
         return ExprError();
+      }
+      /*
```

Comment:
This comment is largely unnecessary and overly wordy.  Also, don't use C style comments anywhere, the Coding standard doesn't want them.

Something simple like, "Diagnose otherwise valid host builtins in device code."